src/gt4py/next/backend.py

```diff
@@ -18,14 +18,119 @@
+    foast_to_itir: workflow.Workflow[ffront_stages.FoastOperatorDefinition, itir.Expr]
+
+    @property
+    def step_order(self) -> list[str]:
+        return [
+            "func_to_foast",
+            "foast_inject_args",
+            "foast_to_past_closure",
+            "past_transform_args",
+            "past_to_itir",
+        ]
+
+
+@dataclasses.dataclass(frozen=True)
+class ProgramTransformWorkflow(workflow.NamedStepSequence):
+    """Modular workflow for transformations with access to intermediates."""
+
+    func_to_past: workflow.SkippableStep[
+        ffront_stages.ProgramDefinition | ffront_stages.PastProgramDefinition,
+        ffront_stages.PastProgramDefinition,
+    ]
+    past_lint: workflow.Workflow[
+        ffront_stages.PastProgramDefinition, ffront_stages.PastProgramDefinition
+    ]
+    past_inject_args: workflow.Workflow[
+        ffront_stages.PastProgramDefinition, ffront_stages.PastClosure
+    ]
+    past_transform_args: workflow.Workflow[ffront_stages.PastClosure, ffront_stages.PastClosure]
+    past_to_itir: workflow.Workflow[ffront_stages.PastClosure, stages.ProgramCall]
+
+
+@dataclasses.dataclass(frozen=True)
+class ProgArgsInjector(workflow.Workflow):
+    args: tuple[Any, ...] = dataclasses.field(default_factory=tuple)
+    kwargs: dict[str, Any] = dataclasses.field(default_factory=dict)
+
+    def __call__(self, inp: ffront_stages.PastProgramDefinition) -> ffront_stages.PastClosure:
+        return ffront_stages.PastClosure(
+            past_node=inp.past_node,
+            closure_vars=inp.closure_vars,
+            grid_type=inp.grid_type,
+            args=self.args,
+            kwargs=self.kwargs,
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class FopArgsInjector(workflow.Workflow):
+    args: tuple[Any, ...] = dataclasses.field(default_factory=tuple)
+    kwargs: dict[str, Any] = dataclasses.field(default_factory=dict)
+    from_fieldop: Any = None
+
+    def __call__(self, inp: ffront_stages.FoastOperatorDefinition) -> ffront_stages.FoastClosure:
+        return ffront_stages.FoastClosure(
+            foast_op_def=inp,
+            args=self.args,
+            kwargs=self.kwargs,
+            closure_vars={inp.foast_node.id: self.from_fieldop},
+        )
+
+
+DEFAULT_FIELDOP_TRANSFORMS = FieldopTransformWorkflow(
+    func_to_foast=func_to_foast.OptionalFuncToFoastFactory(cached=True),
+    foast_inject_args=FopArgsInjector(),
+    foast_to_past_closure=foast_to_past.FoastToPastClosure(
+        foast_to_past=workflow.CachedStep(
+            foast_to_past.foast_to_past,
+            hash_function=eve_utils.content_hash,
+        )
+    ),
+    past_transform_args=past_process_args.past_process_args,
+    past_to_itir=past_to_itir.PastToItirFactory(),
+    foast_to_itir=workflow.CachedStep(
+        step=foast_to_itir.foast_to_itir, hash_function=eve_utils.content_hash
+    ),
+)
+
+
+DEFAULT_PROG_TRANSFORMS = ProgramTransformWorkflow(
```

Comment:
maybe makes sense to put the defaults in the `ProgramTransformWorkflow` definition as you hinted in the overview. Feels cleaner to me. also would have the advantage of being a maker implicitly.

Reply:
Yes, it wouldn't have made sense where they were previously defined, but now it is the way to go!